spec/abstract-operations.html

```diff
@@ -744,4 +744,17 @@ <h1>AddValueToTupleSequenceList ( _sequence_, _value_ )</h1>
       </emu-alg>
     </emu-clause>
   </emu-clause>
+
+  <emu-clause id="sec-isconcatspreadable" aoid="IsConcatSpreadable">
+      <h1>Runtime Semantics: IsConcatSpreadable ( _O_ )</h1>
+      <p>The abstract operation IsConcatSpreadable takes argument _O_. It performs the following steps when called:</p>
+      <emu-alg>
+        1. If Type(_O_) is not Object or Tuple, return *false*
+        1. Let _spreadable_ be ? Get<ins>V</ins>(_O_, @@isConcatSpreadable).
+        1. If _spreadable_ is not *undefined*, return ! ToBoolean(_spreadable_).
+        1. <ins>If Type(_O_) is Tuple or Type(_O_) is Object and _O_ has a [[TupleData]] internal slot, return *true*</ins>
```

Comment:
is there no internal `isTuple` AO for this?

Reply:
not currently, we are manually checking for `TupleData` in the few places we need to. We could certainly add one.

Reply:
I'd prefer that for spec clarity :-)

Reply:
sure thing, just added, also switched `Tuple.isTuple` over to using it.